stack.py

```diff
@@ -72,6 +86,27 @@ def parse_program(code):
             prefix = None
             op = parts[0]
             args = [float(arg) for arg in parts[1:]]
+        if op in sigil_to_op:
+            op = sigil_to_op[op]
+        if op not in valid_ops:
+            raise ValueError("Unknown opcode '{}'".format(op))
+
+        check_type = {
+            int: lambda x: int(x) == x,
+            float: lambda x: isinstance(x, float),
+        }
+        arg_type_list = arg_types[op]
+        typechecked = False
+        for types in arg_type_list:
+            if len(types) != len(args):
```

Comment:
No, since I initially set `typechecked = False`, unless one of those succeeds, it won't accept it. If the arguments differed, then given `[int]`, the typechecker would look at arguments like `[1, 2]`, then `zip` those together, producing `[(int, 1)]`, which would then typecheck, since `zip` stops when the first of its arguments stops.